newdle/client/src/components/creation/userSearch/UserSearch.js

```diff
@@ -19,8 +19,12 @@ async function searchUsers(data, setResults) {
   if (!q.trim()) {
     return;
   }
-  const results = await client.searchUsers(q);
-  setResults(results);
+  try {
+    const results = await client.searchUsers(q);
+    setResults(results);
+  } catch (err) {
+    handleError(err);
```

Comment:
Wouldn't it make sense to include the whole error handling thing directly into the `client` instead?

Reply:
Yeah, and return more structured error data from the server, indicating whether it's something expected or not - check what I'm doing in `require_token` for example.

So whenever you have an error code you can determine how to handle it; and if there's no error code I'd say it's fine to just show the generic error message similar to how you already do.